checkov/common/parsers/yaml/multiline_parser.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import annotations
+
+import re
+from typing import TYPE_CHECKING
+
+from checkov.common.parsers.multiline_parser import BaseMultiLineParser
+
+if TYPE_CHECKING:
+    from detect_secrets.util.code_snippet import CodeSnippet
+
+INDENTATION_PATTERN = re.compile(r'(^\s*(?:-?\s+)?)')
+COMMENT_PREFIX = re.compile(r'^[\s]*(#|\/\/)')
+
+
+class YmlMultilineParser(BaseMultiLineParser):
+
+    def __init__(self) -> None:
+        super().__init__()
+
+    def get_lines_from_same_object(
+            self,
+            search_range: range,
+            context: CodeSnippet | None,
+            raw_context: CodeSnippet | None
+    ) -> set[str]:
+        possible_keywords: set[str] = set()
+        if not context or not raw_context:
+            return possible_keywords
+        for j in search_range:
+            line = context.lines[j]
+            if self.lines_in_same_object(raw_context=raw_context, idx=j) and not self.is_line_comment(line):
+                possible_keywords.add(raw_context.lines[j])
+                if self.is_object_start(raw_context=raw_context, idx=j):
+                    return possible_keywords
+        return possible_keywords
```

Comment:
@marynaKK All of this can be in the shared parent. The implementation of `lines_in_same_object`, `is_line_comment` and `is_object_start` should be per file extension